tensorflow_datasets/image/div2k.py

```diff
@@ -0,0 +1,150 @@
+    "valid_unknown_x2": _DL_URL + "DIV2K_valid_LR_unknown_X2.zip",
+    "train_bicubic_x3": _DL_URL + "DIV2K_train_LR_bicubic_X3.zip",
+    "train_unknown_x3": _DL_URL + "DIV2K_train_LR_unknown_X3.zip",
+    "valid_bicubic_x3": _DL_URL + "DIV2K_valid_LR_bicubic_X3.zip",
+    "valid_unknown_x3": _DL_URL + "DIV2K_valid_LR_unknown_X3.zip",
+    "train_bicubic_x4": _DL_URL + "DIV2K_train_LR_bicubic_X4.zip",
+    "train_unknown_x4": _DL_URL + "DIV2K_train_LR_unknown_X4.zip",
+    "valid_bicubic_x4": _DL_URL + "DIV2K_valid_LR_bicubic_X4.zip",
+    "valid_unknown_x4": _DL_URL + "DIV2K_valid_LR_unknown_X4.zip",
+    "train_bicubic_x8": _DL_URL + "DIV2K_train_LR_x8.zip",
+    "valid_bicubic_x8": _DL_URL + "DIV2K_valid_LR_x8.zip",
+    "train_realistic_mild_x4": _DL_URL + "DIV2K_train_LR_mild.zip",
+    "valid_realistic_mild_x4": _DL_URL + "DIV2K_valid_LR_mild.zip",
+    "train_realistic_difficult_x4": _DL_URL + "DIV2K_train_LR_difficult.zip",
+    "valid_realistic_difficult_x4": _DL_URL + "DIV2K_valid_LR_difficult.zip",
+    "train_realistic_wild_x4": _DL_URL + "DIV2K_train_LR_wild.zip",
+    "valid_realistic_wild_x4": _DL_URL + "DIV2K_valid_LR_wild.zip",
+}
+
+_DATA_OPTIONS = ["bicubic_x2", "bicubic_x3", "bicubic_x4", "bicubic_x8",
+                 "unknown_x2", "unknown_x3", "unknown_x4",
+                 "realistic_mild_x4", "realistic_difficult_x4",
+                 "realistic_wild_x4"]
+
+class Div2kConfig(tfds.core.BuilderConfig):
+  """BuilderConfig for Div2k."""
+
+  def __init__(self, data, **kwargs):
+    """Constructs a Div2kConfig."""
+    if data not in _DATA_OPTIONS:
+      raise ValueError("data must be one of %s" % _DATA_OPTIONS)
+
+    name = kwargs.get("name")
+    if name is None:
+      name = data
+    kwargs["name"] = name
+
+    description = kwargs.get("description")
+    if description is None:
+      description = "Uses %s data." % data
+    kwargs["description"] = description
+
+    super(Div2kConfig, self).__init__(**kwargs)
+    self.data = data
+
+  def download_urls(self):
+    """Returns train and validation download urls for this config."""
+    urls = {
+        "train_lr_url": _DL_URLS["train_"+self.data],
+        "valid_lr_url": _DL_URLS["valid_"+self.data],
+        "train_hr_url": _DL_URLS["train_hr"],
+        "valid_hr_url": _DL_URLS["valid_hr"],
+    }
+    return urls
+
+def _make_builder_configs():
+  configs = []
+  for data in _DATA_OPTIONS:
+    configs.append(Div2kConfig(
+        version=tfds.core.Version("1.0.0"),
+        data=data))
+  return configs
+
+class Div2k(tfds.core.GeneratorBasedBuilder):
+  """DIV2K dataset: DIVerse 2K resolution high quality images"""
+
+  BUILDER_CONFIGS = _make_builder_configs()
+
+  def _info(self):
+    return tfds.core.DatasetInfo(
+        builder=self,
+        description=_DESCRIPTION,
+        features=tfds.features.FeaturesDict({
+            "lr": tfds.features.Image(),
+            "hr": tfds.features.Image(),
+        }),
+        citation=_CITATION,
+    )
```

Comment:
It would be nice if you put in the source of the data by using
`homepage="website from where the data is downloaded"` inside the `DatasetInfo` constructor.

Reply:
Upon adding `homepage=_DL_URL,` returns error `TypeError: __init__() got an unexpected keyword argument 'homepage'`

Reply:
It shouldn't have any error.
Are you sure you are doing it correctly?
If yes, let's ignore that for now. And focus on the other fixes for now.

Reply:
I put it right above the line with the citation, is that supposed to be correct? I will continue working on the other fixes in the meanwhile.